src/AWS/CRT/HTTP/Request.php

```diff
@@ -2,11 +2,16 @@
 
 namespace AWS\CRT\HTTP;
 
+use AWS\CRT\IO\InputStream;
+
 class Request extends Message {
     private $body_stream = null;
 
-    public function __construct($method, $path, $query = [], $headers = null, $body_stream = null) {
+    public function __construct($method, $path, $query = [], $headers = [], $body_stream = null) {
         parent::__construct($method, $path, $query, $headers);
+        if (!is_null($body_stream) && !($body_stream instanceof InputStream)) {
+            throw \InvalidArgumentException('body_stream must be an ' . InputStream::class);
```

Comment:
very nitpicky, but maybe have it say "body_stream must be an instance of" instead, so that if the class name changes then it will still be grammatically correct